src/tools/rustdoc-gui/tester.js

```diff
@@ -101,25 +118,91 @@ async function main(argv) {
     let failed = false;
     let files;
     if (opts["files"].length === 0) {
-        files = fs.readdirSync(opts["tests_folder"]).filter(file => path.extname(file) == ".goml");
+        files = fs.readdirSync(opts["tests_folder"]);
     } else {
-        files = opts["files"].filter(file => path.extname(file) == ".goml");
+        files = opts["files"];
+    }
+    files = files.filter(file => path.extname(file) == ".goml");
+    if (files.length === 0) {
+        console.log("rustdoc-gui: No test selected");
+        process.exit(2);
     }
-
     files.sort();
-    for (var i = 0; i < files.length; ++i) {
+
+    console.log(`running ${files.length} rustdoc-gui tests`);
+    process.setMaxListeners(files.length + 1);
+    let tests = [];
+    let results = new Array(files.length);
+    // poormans enum
+    const RUN_SUCCESS = 42, RUN_FAILED = 23, RUN_ERRORED = 13;
+    for (let i = 0; i < files.length; ++i) {
         const testPath = path.join(opts["tests_folder"], files[i]);
-        await runTest(testPath, options).then(out => {
-            const [output, nb_failures] = out;
-            console.log(output);
-            if (nb_failures > 0) {
+        tests.push(
+            runTest(testPath, options)
+            .then(out => {
+                //console.log(i);
+                const [output, nb_failures] = out;
+                results[i] = {
+                    status: nb_failures === 0 ? RUN_SUCCESS : RUN_FAILED,
```

Comment:
Instead of this, why not have three arrays, one for success, one for failures and one for errors? It even makes it simpler to handle it afterwards.

Reply:
Hm, I guess my mind was very rusty and I emulated an enum :upside_down_face:

Reply:
;)

Reply:
Should we keep it as is? With the separate Arrays I would still have to pre-allocate and then filter afterwards.

Reply:
I'm not sure there is much point to pre-allocate anything. Maybe simply keep the promise around and push it to the "correct" array once it's done?

Reply:
Then to keep the order we would have to keep the initial index and sort all the result arrays

Reply:
That sounds fine to me. :p

Reply:
Or you can store the test name in the array too. Like: `{name: test_name, output: output}` and then you sort on the `name` field.